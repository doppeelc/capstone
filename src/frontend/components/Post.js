import React, { useContext, useEffect, useState } from "react";
import UserApi from "../api";
import UserContext from "./UserContext";
import "../styles/Post.css";


function Post({ post, isLiked }) {

    const { currentUser } = useContext(UserContext);
    const [ isLoading, setIsLoading ] = useState(true);
    const [ liked, setliked ] = useState(false);

    useEffect(() => {
        async function getIsLiked() {
            setliked(isLiked(post.id));
            setIsLoading(false);
        }
        getIsLiked();
    }, [currentUser, post, isLiked]);

    async function toggleLike() {
        if(liked) {
            await UserApi.unLikePost(currentUser.data.username, post.id);
            setliked(false);
        } else {
            await UserApi.likePost(currentUser.data.username, post.id);
            setliked(true);
        }
    }

    if(isLoading) {
        return <p>Loading...</p>;
    }
    
    return (
        <div className="Post" >
            <a className="Post-username" href={`/users/${post.username}`} >
                {post.username}
            </a>
            <p className="Post-content" >
                {post.content}
            </p>
            <p className="Post-time" >
                {post.timePosted.split('.')[0].replace("T", " at ")}
            </p>
            <button onClick={toggleLike} >{ liked ? "Unlike" : "Like" }</button>
        </div>
    )
}

export default Post;