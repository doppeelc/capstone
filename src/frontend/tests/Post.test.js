import { render } from '@testing-library/react';
import Post from '../components/Post';
import UserContext from '../components/UserContext';
import { currentUser, testPost } from './__testCommon';
import { BrowserRouter } from 'react-router-dom/cjs/react-router-dom.min';

test('renders without crashing', () => {
    render(
        <BrowserRouter>
            <UserContext.Provider value={{currentUser}}>
                <Post post={testPost} isLiked={(id) => {return false}} />
            </UserContext.Provider>
        </BrowserRouter>
    );
});

test('matches the snapshot', () => {
    const { asFragment } = render(
        <BrowserRouter>
            <UserContext.Provider value={{currentUser}}>
                <Post post={testPost} isLiked={(id) => {return false}} />
            </UserContext.Provider>
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});






