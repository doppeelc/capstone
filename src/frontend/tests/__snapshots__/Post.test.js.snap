// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="Post"
  >
    <a
      class="Post-username"
      href="/users/testuser"
    >
      testuser
    </a>
    <p
      class="Post-content"
    >
      test post
    </p>
    <p
      class="Post-time"
    >
      stringValue
    </p>
    <button>
      Delete Post
    </button>
    <button>
      Like
    </button>
  </div>
</DocumentFragment>
`;
